src/GAP.jl

```diff
@@ -229,6 +229,10 @@ function __init__()
             ),
         )
     end
+
+    # hack: Let `GAP_jll.MPtr` print itself as `GAP.GapObj`.
+    GAP_jll.MPtr.name.module = GAP
+    GAP_jll.MPtr.name.name = :GapObj
```

Comment:
Maybe defining `Base.show(io::IO, ::Type{GAP_jll.MPtr}) = print(io, "GapObj")` would be slightly less hacky, would it good enough?

Reply:
Thanks @rfourquet .
After `import GAP`, one can access `GAP.GapObj` but not `GapObj`.
After `using GAP`, one can access also `GapObj`.
How can I achieve with your approach that "GAP.GapObj" is printed in the first case, and "GAP" in the latter?

Reply:
Taking inspiration from `Base.show_type_name`:
```julia
function Base.show(io::IO, ::Type{GapObj})
    from = get(io, :module, Main)
    if from === nothing || !Base.isvisible(:GapObj, GAP, from)
        show(io, GAP)
        print(io, ".")
    end
    print(io, "GapObj")
end
```

Reply:
Possibly wrapping the printing of the prefix within `if !(get(io, :compact, false)::Bool)` like in `Base.show_type_name`.

Reply:
Thank you @rfourquet

Reply:
Thanks @rfourquet , I have updated my proposal.